src/vehicleUrban.cpp

```diff
@@ -36,10 +36,18 @@ IntervalCommand *commands[] = { &gpsLat, &gpsLong, &thermoTemp1, &urbanHeadlight
     &urbanLeftSig, &urbanWipers, NULL};
 
 String publishName = "BQIngestion";
+void sendCanSpeed(float speed){
+    CanMessage message = CAN_MESSAGE_NULL; // construct the message and 
+    message.id = CAN_TELEMETRY_GPS_SPEED;
+    message.data[0] = (uint8_t)(speed*3.6);
```

Comment:
Colby made a good point about checking speed to make sure it doesn't go out of bounds of an unsigned 8-bit int [0,255]. Maybe implement a check on speed to make sure it's positive and less than 70? That way it can't end up exceeding the range. 

The speed we get from the GPS library *shouldn't* be negative, but the library is long and complex and I can't necessarily tell if the library code prevents that. It also shouldn't be crazy fast, but with GPS error you never know.

Reply:
I have implemented a check can you please look at it.